### PR TITLE
docs: update `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by angeloashmore**
**Resolves**: DT-2182

### Description

This PR updates `CONTRIBUTING.md` to the new standard Prismic format.

### Checklist

- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [x] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### Preview

https://github.com/prismicio/slice-machine/blob/aa/contributing/CONTRIBUTING.md

### How to QA [^1]

Check the updated `CONTRIBUTING.md` and let me know if it makes sense or if it is missing information. Keep in mind we can always iterate on the document as we come across new findings.

[^1]:
    Please use these labels when submitting a review:
    :question: #ask:&ensp;Ask a question.
    :bulb: #idea:&ensp;Suggest an idea.
    :warning: #issue:&ensp;Strongly suggest a change.
    :tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

**Review by lihbr — APPROVED** [review]
LGTM, it'll be nice to a review from direct DevTools member still 🚀

**Comment by xrutayisire** [comment]
I will do one for sure 🙌

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,56 +1,146 @@
...
+
+- Sentry monitors errors in production.
+- Check Sentry regularly for errors. Especially check Sentry after publishing new package versions.
+
+## :construction_worker: Develop
+
+> [!NOTE]
+> It's highly recommended to discuss your changes with the Prismic team before starting by [opening an issue](https://github.com/prismicio/slice-machine/issues/new/choose).[^2]
+>
+> A short discussion can accellerate your work and ship it faster.
```
Comment (levimykel) [inline CONTRIBUTING.md:51]: #issue Spelling: accelerate

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,56 +1,146 @@
...
+Version increment types can be specified for each package using the action's workflow form. For example, you can choose to increment `slice-machine-ui` by a major, minor, or patch version. Select "decline" to skip incrementing the version.
 
-<!-- Links -->
+The GitHub Action is powered by a purpose-built script: `script/publish.ts`.
 
-[forum-question]: https://community.prismic.io
-[repo-bug-report]: https://github.com/prismicio/slice-machine/issues/new?assignees=&labels=bug&template=bug_report.md&title=
-[repo-feature-request]: https://github.com/prismicio/slice-machine/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=
-[repo-pull-requests]: https://github.com/prismicio/slice-machine/pulls
+[^1]: This package is maintained by the DevTools team. Prismic employees can ask for help or a review in the [#team-dev-tools](https://prismic-team.slack.com/archives/CPG31MDL1) Slack channel.
```
Comment (levimykel) [inline CONTRIBUTING.md:143]: ❓ Should this (and the 2 following slack channel links) reference the `#dev-tools-dev` channel instead?

File: CONTRIBUTING.md (hunk 1 above)
Comment (levimykel) [inline CONTRIBUTING.md]: ⚠️ Spelling: accelerate

**Review by levimykel — APPROVED** [review]
LGTM.  I left one question and a quick spelling fix, otherwise I approve 🙌

File: CONTRIBUTING.md (hunk 2 above)
Comment (xrutayisire) [reply CONTRIBUTING.md:143]: Hum, I'm not even sure we should mention a private internal link, WDYT?

Comment (levimykel) [reply CONTRIBUTING.md:143]: Yeah, since this is a public repo, we should remove the links. We could keep the unlinked channel names, though.

Comment (levimykel) [reply CONTRIBUTING.md:143]: We decided on slack to keep the links as is 😄

**Pull request merged**